File: api-geteway/src/main/java/br/com/isis/apigateway/ApiGetewayApplication.java
package br.com.isis.apigateway;

import com.netflix.zuul.ZuulFilter;
import com.netflix.zuul.context.RequestContext;
import org.springframework.boot.SpringApplication;
import org.springframework.boot.autoconfigure.SpringBootApplication;
import org.springframework.cloud.client.discovery.EnableDiscoveryClient;
import org.springframework.cloud.netflix.zuul.EnableZuulProxy;
import org.springframework.core.annotation.Order;
import org.springframework.stereotype.Component;

import java.util.Set;

@SpringBootApplication
@EnableZuulProxy
@EnableDiscoveryClient
public class ApiGetewayApplication {

	public static void main(String[] args) {
		SpringApplication.run(ApiGetewayApplication.class, args);
	}

	@Order(2)
	@Component
	@SuppressWarnings("unchecked")
	public class CorrectHeadersFilter extends ZuulFilter {

		@Override
		public String filterType() {
			return "pre";
		}

		@Override
		public int filterOrder() {
			return 10000;
		}

		@Override
		public boolean shouldFilter() {
			return true;
		}

		@Override
		public Object run() {
			RequestContext currentContext = RequestContext.getCurrentContext();

			currentContext.addZuulRequestHeader("host",  currentContext.getRequest()
					.getHeader("host"));

			// XXX
			currentContext.getZuulRequestHeaders().remove("x-forwarded-prefix");

			((Set<String>) currentContext.get("ignoredHeaders")).clear();// XXX
			return null;
		}

	}
}

File: api-geteway/src/main/kotlin/br/com/isis/apigateway/ApiGetewayApplication.kt
package br.com.isis.apigateway

import org.springframework.boot.SpringApplication
import org.springframework.boot.autoconfigure.SpringBootApplication
import org.springframework.cloud.client.discovery.EnableDiscoveryClient
import org.springframework.cloud.netflix.zuul.EnableZuulProxy

/**
 * Created by gaspar on 08/05/2017.
 */
@SpringBootApplication
@EnableZuulProxy
@EnableDiscoveryClient
class ApiGetewayApplication

    fun main(args: Array<String>) {
        SpringApplication.run(ApiGetewayApplication::class.java, *args)
    }
